spicelib/raw/raw_read.py

```diff
@@ -339,189 +360,428 @@ def namify(spice_ref: str):
         raise NotImplementedError(f'Unrecognized alias type for alias : "{spice_ref}"')
 
 
-class RawRead(object):
-    """Class for reading Spice wave Files. It can read all types of Files. If stepped data is detected,
-    it will also try to read the corresponding LOG file so to retrieve the stepped data.
+# base class that has the minimum for a plot. Inherited by the PlotData and RawRead classes
+class PlotInterface(ABC):
+    
+    @property
+    @abstractmethod
+    def nVariables(self) -> int:
+        """Number of variables in the RAW file
+        """
+        pass
 
-    :param raw_filename: The file containing the RAW data to be read
-    :type raw_filename: str | pahtlib.Path
-    :param traces_to_read:
-        A string or a list containing the list of traces to be read. If None is provided, only the header is read and
-        all trace data is discarded. If a '*' wildcard is given or no parameter at all then all traces are read.
-    :type traces_to_read: str, list or tuple
-    :param dialect: The simulator used. 
-        Please use from ["ltspice","qspice","ngspice","xyce"]. If not specified, dialect will be auto detected. 
-        This is likely only needed for older versions of ngspice and xyce. ltspice and qspice can reliably be auto detected.
-    :type dialect: str    
-    :key headeronly:
-        Used to only load the header information and skip the trace data entirely. Use `headeronly=True`.
-    """
-    header_lines = (
-        "Title",
-        "Date",
-        "Plotname",
-        "Output",
-        "Flags",
-        "No. Variables",
-        "No. Points",
-        "Offset",
-        "Command",
-        "Variables",
-        "Backannotation"
-    )
-
-    ACCEPTED_PLOTNAMES = (
-        'AC Analysis',
-        'DC transfer characteristic',
-        'Operating Point',
-        'Transient Analysis',
-        'Transfer Function',
-        'Noise Spectral Density',
-        'Frequency Response Analysis',
-    )
-
-    def __init__(self, raw_filename: str, traces_to_read: Union[str, List[str], Tuple[str, ...]] = '*', dialect: str = None, **kwargs):
-        self.dialect = None
-        """The dialect of the spice file read. This is either set on init, or detected """
-        
-        self.verbose = kwargs.get('verbose', True)
+    @property
+    @abstractmethod
+    def nPoints(self) -> int:
+        """Number of points in the RAW file
+        """
+        pass
+    
+    @property
+    @abstractmethod
+    def raw_type(self) -> str:
+        """The type of the RAW file, either 'binary:' or 'values:'"""
+        pass
+    
+    @property
+    @abstractmethod
+    def aliases(self) -> Dict[str, str]:
+        """QSpice defines aliases for some of the traces that can be computed from other traces.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def backannotations(self) -> List[str]:
```

Comment:
Minor nitpick. list and dict are now genertic types in Python 3.9+. This means you can replace `List` with the built-in `list` and avoid importing `List` from the `typing `module.

Same applies to `Dict`, you can use lower case `dict` instead.

See https://peps.python.org/pep-0585/